andela/firestar-backend
# 167730742 Ft login via email and password

#### What does this PR do?
* Implement login feature via email and password
#### Description of Task to be completed?
* N/A
#### How should this be manually tested?
* git clone `https://github.com/andela/firestar-backend.git`
* cd `firestar-backend`
* Checkout to the branch `git checkout ft-login-email-password-167730742`
* run `npm install`
* Setup test databases, see `.env.example` for sample 
**P.S** `JWT_SECRET_KEY=pass a string of characters e.g THISISBAREFOOTNOMADPROJECT`
* Run migration to create tables 
         - run `npm run migrate:dev` for development
        -  run `npm run migrate:test` for test
* Populate test database: ` npm run seed:test`
* To run the test  `npm run test`

* Start the sever by runnning `npm run start:dev`

* Test the endpoint `localhost:3000/api/v1/users/auth/login` with  **postman** ,  then select POST as the request with the following as the body:
     ```
    {
      email: 'example@gmail.com',
      password: 'firestar2019@K`
     }
    {
     email: 'example1@gmail.com',
     password: 'firestar2019@K'
     }
      
     {
      email: 'youremail20@andela.com',
      password: 'firestar2019@K'
     }

**P.S** use only the above user details to login because these are some of the data that will be seeded to the database after running ` npm run seed:dev`
          
* Successful login will output the following below:
![Screenshot from 2019-08-29 21-56-19](https://user-images.githubusercontent.com/22738789/63975886-f75e1c80-caa7-11e9-8f0a-f6c9cfc4dc57.png)
#### Any background context you want to provide?
* N/A
#### What are the relevant pivotal tracker stories?
* [#167730742](https://www.pivotaltracker.com/story/show/167730742)
#### Screenshots or gifs (if appropriate)
![Screenshot from 2019-08-29 18-06-07](https://user-images.githubusercontent.com/22738789/63975907-06dd6580-caa8-11e9-876a-21eb98967769.png)
#### Questions:
* N/A
